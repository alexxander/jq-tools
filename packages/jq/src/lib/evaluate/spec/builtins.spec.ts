import { expectCode, expectCodeError } from './specUtils.spec';

describe('builtins', () => {
  describe('native', () => {
    describe('length', () => {
      it('null', () => {
        expectCode('null | length', [0]);
      });
      it('boolean', () => {
        expectCodeError('true | length');
      });
      it('number', () => {
        expectCodeError('5 | length');
      });
      it('string', () => {
        expectCode('"abc" | length', [3]);
      });
      it('array', () => {
        expectCode('[1,2,3,4,5] | length', [5]);
      });
      it('object', () => {
        expectCode('{"a": 1, "b": 2} | length', [2]);
      });
      it('multi', () => {
        expectCode(
          '"abcdef", [1,2,3], {"a": 1, "b": 2, "c": 3}, "", [], {} | length',
          [6, 3, 3, 0, 0, 0]
        );
      });
    });
  });
});
