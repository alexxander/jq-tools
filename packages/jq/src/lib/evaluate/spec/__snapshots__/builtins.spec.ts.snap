// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`builtins native length boolean 1`] = `"boolean has no length"`;

exports[`builtins native length number 1`] = `"number has no length"`;
